File: JZ Offer/1.cpp
class Solution {
public:
	bool Find(int target, vector<vector<int> > array) {
		int rows = (int)array.size();
		int cols = (int)array[0].size();
		int j = cols - 1;
		int i = 0;
		while (j >= 0 && i <= rows - 1)
		{
			if (array[i][j] > target)
			{
				j--;
			}
			else if (array[i][j] < target)
			{
				i++;
			}
			else
			{
				return 1;
			}
		}
		return 0;
	}
};

File: Leetcode/1.py
# -*- coding: utf-8 -*-
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dictnum = {}
        for i in range(len(nums)):
            if target-nums[i] in dictnum:
                return [dictnum[target-nums[i]],i]
            dictnum[nums[i]] = i
